cluster-provision/images/vm-image-builder/create-containerdisk.sh

```diff
@@ -1,95 +1,126 @@
 #!/usr/bin/env bash
 set -exuo pipefail
 
-
 if [ "$#" -ne 1 ]; then
     echo "Usage: create-containerdisk.sh image-directory"
     echo "Run `create-continerdisk.sh example` to build the `example` image in the `example` folder"
 fi
 
 SCRIPT_PATH=$(dirname "$(realpath "$0")")
+source ${SCRIPT_PATH}/common.sh
+ARCH="${ARCHITECTURE:-"$(go_style_local_arch)"}"
 
 export CUSTOMIZE_IMAGE_SCRIPT=${CUSTOMIZE_IMAGE_SCRIPT:-"${SCRIPT_PATH}/customize-image.sh"}
 
+function download_base_image() {
+    local -r arch=$1
+    local -r image_name=$2
+
+    local url
+    local file_name
+    if [[ ${ARCH} = "amd64" ]]; then
+        url="$(cat ${SCRIPT_PATH}/${IMAGE_NAME}/image-url)"
+        file_name="${image_name}-image.qcow2"
+    else
+        image_url_path=${SCRIPT_PATH}/${IMAGE_NAME}/image-url-${ARCH}
+        url="$(cat ${image_url_path})"
+        file_name="${image_name}-image-${ARCH}.qcow2"
+    fi
+
+    if ! [ -e "${file_name}" ]; then
+        # Download base VM image
+        curl -L "${url}" -o "${file_name}"
+    fi
+
+    echo "${file_name}"
+}
+
 function customize_image() {
-  local source_image=$1
-  local os_variant=$2
-  local customized_image=$3
-  local cloud_config=$4
+    local source_image=$1
+    local os_variant=$2
+    local customized_image=$3
+    local cloud_config=$4
 
-  # Backup the VM image and pass copy of the original image
-  # in case customizing script fail.
-  vm_image_copy="$(dirname "${customized_image}")/copy-${source_image}"
-  cp "${source_image}" "${vm_image_copy}"
+    # Backup the VM image and pass copy of the original image
+    # in case customizing script fail.
+    vm_image_copy="$(dirname "${customized_image}")/copy-${source_image}"
+    cp "${source_image}" "${vm_image_copy}"
 
-  # TODO: convert this script and its dependencies to container
-  ${CUSTOMIZE_IMAGE_SCRIPT} "${vm_image_copy}" "${os_variant}" "${customized_image}" "${cloud_config}"
+    # TODO: convert this script and its dependencies to container
+    ${CUSTOMIZE_IMAGE_SCRIPT} "${vm_image_copy}" "${os_variant}" "${customized_image}" "${cloud_config}"
 
-  # Backup no longer needed.
-  rm -f "${vm_image_copy}"
+    # Backup no longer needed.
+    rm -f "${vm_image_copy}"
+}
+
+function build_container() {
+    local -r build_directory=$1
+    local -r new_vm_image_name=$2
+    local -r arch=$3
+    local -r image_name=$4
+
+    if [[ $arch = $(go_style_local_arch) ]]; then
+        docker_build="docker build . -t ${image_name}:${arch}"
+    else
+        echo "When performing cross builds, make sure 'docker buildx' is installed."
+        docker_build="docker buildx build --platform "linux/${arch}" . -t ${image_name}:${arch}"
```

Comment:
It would be really great if we could use something like `buildah` for the multi-arch image builds and manifests but maybe we can take that in a follow on.

Reply:
Ok. I have verified the current script in `quay.io/kubevirtci/vm-image-builder:v20230607-9021afd`, it works fine.